File: src/main/java/com/iminling/core/util/ResponseWriter.java
package com.iminling.core.util;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.iminling.model.common.ResultModel;
import com.iminling.model.exception.AuthorizeException;

import javax.servlet.http.HttpServletResponse;

public final class ResponseWriter {

    private ResponseWriter() {}

    public static void write(HttpServletResponse response, AuthorizeException ex) throws Exception {
        // HttpServletResponse.SC_FORBIDDEN
        response.setStatus(ex.getMessageCode().getCode());
        response.setCharacterEncoding("UTF-8");
        response.setContentType("application/json;charset=UTF-8");
        response.setHeader("Access-Control-Allow-Origin","*");
        ObjectMapper objectMapper = new ObjectMapper();
        response.getWriter().write(objectMapper.writeValueAsString(ResultModel.isFail(ex.getMessageCode())));
    }
}

File: iminling-core/src/main/kotlin/com/iminling/core/util/ResponseWriter.kt
package com.iminling.core.util

import com.fasterxml.jackson.databind.ObjectMapper
import com.iminling.core.config.value.ResultModel
import com.iminling.core.exception.AuthorizeException
import javax.servlet.http.HttpServletResponse

/**
 * @author  yslao@outlook.com
 * @since  2021/11/26
 */
class ResponseWriter {

    companion object {
        @Throws(Exception::class)
        fun write(response: HttpServletResponse, ex: AuthorizeException) {
            // HttpServletResponse.SC_FORBIDDEN
            response.status = ex.messageCode.code
            response.characterEncoding = "UTF-8"
            response.contentType = "application/json;charset=UTF-8"
            response.setHeader("Access-Control-Allow-Origin", "*")
            val objectMapper = ObjectMapper()
            response.writer.write(objectMapper.writeValueAsString(ResultModel.fail(ex.messageCode)))
        }
    }

}